### ISSUE TITLE
v3.1.2 Page numbers become out of sync

### ISSUE THREAD

**Issue by rayfoss** [FranckFreiburger/vue-pdf#47]
This is not page #1 (Chrome 62) 
https://fiddle.jshell.net/fossfiddler/5k4ptmqg/152/show/light/
![screen shot 2018-01-08 at 5 30 37 pm](https://user-images.githubusercontent.com/876076/34697803-65ffe9ce-f49a-11e7-9b90-e156c702a122.png)

Also the first page doesn't render on iphone emulator iOS 11 iPhone 6, or android emulator N5 API 26. On iphone I can still flip the page and it works well afterwards.
![screen shot 2018-01-08 at 5 03 50 pm](https://user-images.githubusercontent.com/876076/34697830-8f9a9aea-f49a-11e7-89b7-5be5ed174270.png)
![screen shot 2018-01-08 at 4 32 25 pm](https://user-images.githubusercontent.com/876076/34697837-93ff4ba8-f49a-11e7-8a55-eff23953edb8.png)

**Comment by dcean** [comment]
#### I test your jsfiddle by chrome:

#### Using macbookpro, pc side looks great:
![image](https://user-images.githubusercontent.com/21835160/34701225-eb1b7830-f521-11e7-9d31-eb880397388f.png)

### Using chrome mobile device, the result as follows:
![image](https://user-images.githubusercontent.com/21835160/34701304-74881722-f522-11e7-8cd7-1a4766fe08fe.png)

![image](https://user-images.githubusercontent.com/21835160/34701354-b48d1afc-f522-11e7-8f45-976f7d643de5.png)
#### It still blurred......

#### When I test my own project using v3.1.2 by chrome mobile device, I use the demo: `Example - display multiple pages of the same pdf document`
![image](https://user-images.githubusercontent.com/21835160/34701569-fd14acee-f523-11e7-971c-f5f9a058b065.png)
#### It doesn`t work

#### It can be reproduced by using chrome mobile device.
#### We may also have some bugs on mobile side. 

I want to show pdf on mobile phone. The support for mobile side is necessary.

**Comment by rayfoss** [comment]
The second screenshot (11/14) has the wrong page. I wasn't able to reproduce the blurryness, it might be how the GPU handles under stress. This fiddle comments out the "width 75%" style.

https://fiddle.jshell.net/5k4ptmqg/158/

I was able to reproduce the first page not loading bug on a 4k monitor zoomed to 500%... so it's not an explicitly mobile bug. It also shows no blurriness when doing so. But that's probably another issue to tackle separately.

![screenshot from 2018-01-09 06-27-12](https://user-images.githubusercontent.com/876076/34721053-d61b4b56-f506-11e7-9863-a8e767f83465.png)
![screenshot from 2018-01-09 06-28-09](https://user-images.githubusercontent.com/876076/34721054-d63baad6-f506-11e7-8cbd-c035dbb1edf8.png)

**Comment by dcean** [comment]
So we have three problems:
1:  Page numbers become out of sync
2:  Rasterize on mobile (from jsfiddle demo, blurred when pinch to zoom on chrome mobile device  but clear on pc device)
3:  Can`t display on WebApp (from chrome mobile device)

**Comment by rayfoss** [comment]
vector rendering has never been a feature so I would categorize that as an enhancement for now. The priority right now should be fixing the not displaying of the first page, which I've had happen on desktop even at 125% zoom. 

Debounce is a workaround for page numbers becoming out of sync. It seems to happene on edge cases and on slow taxed devices.

**Comment by adminlte** [comment]
> the first page doesn't render 

the same mistake with me in the chrome mobile device when the pdf file is too large 
but when i gone to the page 2 and return to the page 1, this page is successfully rendered.
what happen ? and how should I avoid it ?

**Comment by dcean** [comment]
You can use V3.0.1 before author fix the bug, but pdf may be blurred on mobile.

Or you can use other lib such as pdf.js

**Comment by rayfoss** [comment]
@dcean @adminlte no point in using pdf.js directly on vue... the work in adapting it to play well would be best spent on improving this, which is almost entirely an adapter for pdf.js. The key issue is the worker, which makes keying the vue component a pain. I've tried restarting the component on errors by changing the key programmatically, that's unfortunately not yet feasible...

It's a very small component feel free to submit pull requests... e2e testing that works in travis/gitlab-ci would be a godsend.

**Comment by andreasvirkus** [comment]
Reverting to `v3.0.1` worked for me (for now), thanks @dcean

**Comment by andreasvirkus** [comment]
@rayfoss @dcean, do you happen to know if there is anyone working on this issue (particularly rasterization and not being displayed on mobile devices)?

BR,
Andreas

**Comment by FranckFreiburger** [comment]
please try v3.1.5

**Comment by andreasvirkus** [comment]
Doesn't render the pdf for me on iOS Safari (works on Android Chrome now)

**Comment by FranckFreiburger** [comment]
is there any error message in the console ?

**Comment by andreasvirkus** [comment]
No. Forgot to mention that only the first page is displayed (but PDF optimisation doesn't work also when zooming)

**Comment by andreasvirkus** [comment]
Another issue now seems to be when building my Webpack bundle, then `babel-loader` complains about pdf.js/dist file's dynamic `import()` token:

```
build	02-Feb-2018 13:33:20	ERROR in ./node_modules/babel-loader/lib!./node_modules/vue-loader/lib/selector.js?type=script&index=0!./node_modules/vue-pdf/src/pdf.vue
build	02-Feb-2018 13:33:20	Module build failed: SyntaxError: Unexpected token (186:10)
build	02-Feb-2018 13:33:20	
build	02-Feb-2018 13:33:20	  184 | 		this.fetch = function(query) {
build	02-Feb-2018 13:33:20	  185 | 
build	02-Feb-2018 13:33:20	> 186 | 			return import('raw-loader!pdfjs-dist/cmaps/'+query.name+'.bcmap' /* webpackChunkName: "noprefetch-[request]" */)
build	02-Feb-2018 13:33:20	      | 			       ^
build	02-Feb-2018 13:33:20	  187 | 			.then(function(bcmap) {
build	02-Feb-2018 13:33:20	  188 | 
build	02-Feb-2018 13:33:20	  189 | 				return {
build	02-Feb-2018 13:33:20	
build	02-Feb-2018 13:33:20	 @ ./node_modules/vue-pdf/src/pdf.vue 7:2-87
```

**Comment by FranckFreiburger** [comment]
@andreasvirkus, please fill a new issue about dynamic import()

**Comment by andreasvirkus** [comment]
Sure thing!
Done https://github.com/FranckFreiburger/vue-pdf/issues/55

**Comment by andreasvirkus** [comment]
Any progress on this or any way I could help out with debugging?
Only the first page is still rendered for me with `3.2.0`. Not only on mobile devices,
but also on Chrome's responsive view on desktop

![screen shot 2018-02-07 at 12 05 11](https://user-images.githubusercontent.com/9140811/35910465-4bbc3370-0bff-11e8-86a2-edf51c0ce63e.png)

**Comment by FranckFreiburger** [comment]
@andreasvirkus, Please open a new issue for this bug
